TaylorFramework/Modules/scissors/ExtendedComponent.swift

```diff
@@ -98,6 +98,11 @@ final class ExtendedComponent {
         }
     }
     
+    func filter(type: ComponentType) {
+        components = components.filter { $0.type != type }
+        for component in components { component.filter(type) }
```

Comment:
We can use `components.forEach { $0.filter(type) }` here to make it more expressive.